# HG changeset patch
# User Kai T. Ohlhus <k.ohlhus@gmail.com>
# Date 1471357380 -7200
#      Tue Aug 16 16:23:00 2016 +0200
# Node ID 2cd1afd0f12f3259a4bee14dfcbfd44d6ec10ad1
# Parent  510886d03ef2e0a9db03e80dbc522ff38471424d
publish.m: prevent formatting of inline and block math.

diff --git a/scripts/general/publish.m b/scripts/general/publish.m
--- a/scripts/general/publish.m
+++ b/scripts/general/publish.m
@@ -807,26 +807,36 @@ function str = format_text (str, formatt
   ## TODO: better pointer to the function documentation
   str = regexprep (str, '<octave:([^\s<>]*) *([^<>$]*)>', ...
     formatter ("link", ["https://www.gnu.org/software/octave/", ...
       "doc/interpreter/Function-Index.html"], "$2"));
   ## Links "<http://www.someurl.com TEXT>"
   str = regexprep (str, '<(\S{3,}[^\s<>]*) *([^<>$]*)>', ...
     formatter ("link", "$1", "$2"));
   oldstr = str;
+  ## Save inline "$" and block "$$" LaTeX math
+  [~,~,~,math_cstr] = regexp (str, '\${1,2}.*?\${1,2}');
+  for i = 1:length(math_cstr)
+    str = regexprep (str, '\${1,2}(.*?)\${1,2}', ...
+      ["PUBLISHMATH", num2str(i)], "once");
+  endfor
   ## Loop because of inlined expressions, e.g. *BOLD _ITALIC_*
   do
     oldstr = str;
     ## Bold
     str = regexprep (str, '\*([^*$_|]*)\*', formatter ("bold", "$1"));
     ## Italic
     str = regexprep (str, '_([^_$|*]*)_', formatter ("italic", "$1"));
     ## Monospaced
     str = regexprep (str, '\|([^|$_*]*)\|', formatter ("monospaced", "$1"));
   until (strcmp (str, oldstr))
+  ## Restore inline "$" and block "$$" LaTeX math
+  for i = 1:length(math_cstr)
+    str = strrep (str, ["PUBLISHMATH", num2str(i)], math_cstr{i});
+  endfor
   ## Replace special symbols
   str = strrep (str, "(TM)", formatter("TM"));
   str = strrep (str, "(R)", formatter("R"));
 endfunction
 
 
 
 function doc_struct = eval_code (doc_struct, options)
